ctapipe/io/tests/test_hdf5.py

```diff
@@ -28,24 +32,45 @@ def test_write_container(temp_h5_file):
     r0tel.meta['test_attribute'] = 3.14159
     r0tel.meta['date'] = "2020-10-10"
 
-    writer = HDF5TableWriter(str(temp_h5_file), group_name='R0',
-                             filters=tables.Filters(
-        complevel=7))
-    writer.exclude("tel_002", ".*samples")  # test exclusion of columns
+    with HDF5TableWriter(
+            str(temp_h5_file),
```

Comment:
is `str(temp_h5_file)` really needed here? Looks like `temp_h5_file` is already a string.

Also the name `temp_h5_file` looks wrong to me ... it is a path not a file-like

Reply:
I also do not like the whole test setup here. Tests depend on each other.The first test writes into the same temporary file as another reads. Order is important here. Parallel execution of tests might not be possible, right?

Reply:
I agree. That's my fault - the tests were poorly written. We should separate them so that they do not depend on each other

Reply:
the `str(temp_h5_file)` was just to get the filename as a string, since pytables wants just a string, not whatever the tmpfile fixture gives back (which is perhaps a pathlib Path?). There's probably a better way.

Reply:
done